### ISSUE TITLE
category of search changes unexpectedly

### ISSUE THREAD

**Issue by privacyguy123** [searxng/searxng#2859]
Perhaps someone can look into this bug? It's driving me loopy

Exact steps to reproduce:

- Search for something from within the URLbar with SearXNG set as default search engine in FireFox
- Results appear with saved preferences
- Change the search term within the "Search for" box and click the magnifying glass
- Results default to just "IT" selected for no apparent reason???
- Further digging shows this only happens with ```Search on category select - ON``` - I would like this option to work properly.

**Comment by Bnyro** [comment]
No, since everybody here is doing their work voluntary, nobody has to look at this.

**Comment by privacyguy123** [comment]
> No, since everybody here is doing their work voluntary, nobody has to look at this.

Let me rephrase

**Comment by return42** [comment]
Shouldn't this issue be fixed by:

- https://github.com/searxng/searxng/pull/2840

What instance do you use?

**Comment by privacyguy123** [comment]
> Shouldn't this issue be fixed by:
> 
> * [[fix] magnifier button always search in the general category (#2839) #2840](https://github.com/searxng/searxng/pull/2840)
> 
> What instance do you use?

It does sound related yes ... I have a private Docker instance. Perhaps the code for the update hasn't been pushed over there? 

There's another similar bug that feels as if t's FireFox cache serving too fast or something and NO preferences get read at all and I get an unthemed search in "General" until page refresh. Very strange.

**Comment by return42** [comment]
> Search for something from within the URLbar with SearXNG set as default search engine in FireFox
> There's another similar bug that feels  ... and I get an unthemed search in "General" until page refresh. Very strange.

There is a connection between the search via the URL bar and the preferences which is not obvious and can end in inexplicable behavior:

If you add a SearXNG as a search engine to your WEB browser, then the preferences that you have saved at the time will be stored in this URL bar search (at least partially).

These preferences will be applied from now on whenever you search via the URL-bar.

If you change your preferences or if we change the scheme of the preferences (like it happened with PR https://github.com/searxng/searxng/pull/2740), this can lead to unpredictable behavior, because the preferences that are then sent to the SearXNG instance might be a bit messed up.

My recommendation would be to remove the SearXNG instance in the WEB browser, reset all preferences and then add the instance again to the WEB browser as search engine --> "URL bar without preferences".

Alternatively, remove the SearXNG instance in the WEB browser, set preferences as desired, and then add the instance back to the WEB browser as a search engine --> "URL bar with preferences".

The "URL bar with preferences" can lead into unpredictable behavior, next time SearXNG will change its preferences scheme in a manner that is not downward compatible.

----

Can you please test if it helps, when you remove & re-add the instance as search engine in your WEB browser ..

----

UPDATE: might related to ..

- https://github.com/searxng/searxng/pull/2856

**Comment by privacyguy123** [comment]
Going through these steps to delete and re-add SearXNG as default with no preferences saved first of all reproduces the same bug except now "News" is highlighted by default for no apparent reason.

Directly correlated with ```Search on category select - ON``` seems to go away with this option off.

**Comment by return42** [comment]
hm, strange / thanks for testing .. is the instance public ... could you post the URL here?

**Comment by privacyguy123** [comment]
> hm, strange / thanks for testing .. is the instance public ... could you post the URL here?

SearXNG-docker official image on Pi 4B running latest DietPi. I ran ```docker compose pull``` before this testing incase any updates were pushed that I missed.

**Comment by BernieHuang2008** [comment]
Did you test it on other instances Except your Pi, so that I can reproduce it?
(Or maybe there's another way to reproduce?)

**Comment by privacyguy123** [comment]
> Did you test it on other instances Except your Pi, so that I can reproduce it? (Or maybe there's another way to reproduce?)

I've no reason to use public instances so I have not tested them. Steps to reproduce are outlined in first post, I can get it to happen _every_ time with the ```Search on category select - ON``` - it's sporadic with that option set to ```OFF``` ... Sometimes I get a fully unthemed page with no preferences, sometimes I get a random category selected. 

@Bnyro you really just follow this issue to thumbs down all the posts? Sad. 

EDIT: same steps reproduce the same bug on ```https://searxng.nicfab.eu/searxng/search``` - "News" is the only thing highlighted randomly after saving preferences that are not that.

**Comment by BernieHuang2008** [comment]
sorry, but i can't produce it.
my environment: `latest firefox`+`public instance (https://search.demoniak.ch)`

does anyone reproduced it successfully? 
if none did and you can't provide us your instance, we may not be able to solve it ...

**Comment by privacyguy123** [comment]
> sorry, but i can't produce it. my environment: `latest firefox`+`public instance (https://search.demoniak.ch)`
> 
> does anyone reproduced it successfully? if none did and you can't provide us your instance, we may not be able to solve it ...

Perhaps a FireFox setting and not a SearXNG issue specifically ... are settings being stored anywhere else except the cookie? I have a modded FireFox user.js setup for privacy, it deletes a lot of info on browser close ... I have obviously forced allowed the cookie that the preferences are saved in but that doesn't explain why it'll _sometimes_ respect the preferences and sometimes not, especially while toggling ```Search on category select```

(obviously, but just for completeness) Pi address is added to exceptions in FireFox 
![image](https://github.com/searxng/searxng/assets/50337995/8abe3c8e-98a8-469b-8c83-2f5e0dabc18e)

EDIT: Also for completeness, same bug with FireFox in "troubleshoot mode."

**Comment by return42** [comment]
> are settings being stored anywhere else except the cookie?

no

> I have a modded FireFox user.js setup for privacy, it deletes a lot of info on browser close ..

same on my side .. I delete all data that is stored in ffox on closing (include deletion of all cookies).

What for me and @BernieHuang2008 is not clear: is it just on your instance or can you reproduce the issue on other instances?

**Comment by privacyguy123** [comment]
Same steps reproduce the same bug on https://searxng.nicfab.eu/searxng/search - "News" is the only thing highlighted randomly after saving preferences that are not that.

**Comment by BernieHuang2008** [comment]
sorry, it seems your problem because i can't reproduce it on the instance you offered.
you just said that you have enabled some third-party user-scripts, maybe you can disabled them and test again.

**Comment by privacyguy123** [comment]
Same steps reproduce same bug on ```search.demoniak.ch``` - "News" is randomly highlighted.

I mentioned I tried on FireFox in troubleshoot mode - this means with all addons off. Unless you mean a fresh FireFox profile with no user.js modifications? This makes no sense as your colleague runs a similar setup ...

**Comment by spectrapulse** [comment]
> Same steps reproduce same bug on `search.demoniak.ch` - "News" is randomly highlighted.
> 
> I mentioned I tried on FireFox in troubleshoot mode - this means with all addons off. Unless you mean a fresh FireFox profile with no user.js modifications? This makes no sense as your colleague runs a similar setup ...

Modifying the user.js in Firefox can lead to all kinds of weird side-effects may it be done properly or improperly. So yes, you should be using a fresh Firefox profile to see if the issues you encounter also appear there.

**Comment by return42** [comment]
This thread become more and more a discussion about your firefox setup / We can't reproduce the issue you reported, if you can't reproduce it with a different browser or a fresh install or whatever, its not related to SearXNG .. here we are in the SearXNG issue tracker .. I close this issue now.

**Comment by privacyguy123** [comment]
I can literally reproduce it on a fresh FireFox profile with no addons ...

**Comment by privacyguy123** [comment]
> > Same steps reproduce same bug on `search.demoniak.ch` - "News" is randomly highlighted.
> > I mentioned I tried on FireFox in troubleshoot mode - this means with all addons off. Unless you mean a fresh FireFox profile with no user.js modifications? This makes no sense as your colleague runs a similar setup ...
> 
> Modifying the user.js in Firefox can lead to all kinds of weird side-effects may it be done properly or improperly. So yes, you should be using a fresh Firefox profile to see if the issues you encounter also appear there.

It's done properly and only shows a side effect with saving SearXNG preferences it seems.

**Comment by cy8aer** [comment]
I have this Issue for social media tab. I tested it with official searxng container based on c3ab49cd903d27905d2da6f70699a55c9a74593e

Preparation

* Set "Search when Category clicked" in UI settings
* search something e.g. "catwuschel"
* click "Social Media" - catwuschel  is searched in "Social Media"
* Enter something other e. g. "caturday"  in the search bar and [Enter] 

Expected:

"caturday" is searched in "Social Media"

Actual: 

It is searched in General Tab

**Comment by return42** [comment]
I can reproduce what @cy8aer reported in https://github.com/searxng/searxng/issues/2859#issuecomment-1751779193

The issue is related to the whitespace in the category "social media" .. somewhere in the code its reduced to "social" ... since such a "social" category does not exists (no engine is configured in that category), SearXNG falls back to the general search tab.

**Comment by return42** [comment]
Can't reproduce this issue any more ..
